atsign-foundation/at_widgets
# Fix for #139 and Lint fixes.

### What I did
- Fixed all the null safe issues.
- Fixed unnecessary type castings issue #139.
- Added strict DataTypes as of lint rules.
- Removed `pedantic` as it stated it was [deprecated ](https://pub.dev/packages/pedantic#:~:text=This%20package%20is%20deprecated.).
- Added [lints](https://pub.dev/packages/lints) dev dependency.
- Fixed [autobug.yaml](https://github.com/atsign-foundation/at_widgets/blob/trunk/.github/workflows/autobug.yaml#L16-L17) `if` condition.

### How I did it

- Added lint rules to maintain Strict typing and Typecasting.
- Read [docs about auto bug git actions](https://github.com/srggrs/assign-one-project-github-action#organisation-or-user-project).

### How to verify it

- Not pretty sure how to use test files regarding packages. I would request @sarika01 to review the test and make necessary changes and @DivyaYadavareddy to check the packages once.
- Also not pretty sure how to test git actions locally, I would request @cpswan to review the script(I'm pretty sure it must work).

### Description for the changelog

- Fixed #139 and the code is ready to develop further. 

**NOTE:** Make sure of reviewing correctly. Cause may find some breakings due to Strict typing. Uncomment the lint rules to use strict typing from `analysis_options.yaml`.
